components/support/utils/src/main/java/mozilla/components/support/utils/URLStringUtils.kt

```diff
@@ -28,4 +28,10 @@ object URLStringUtils {
         }
         return uri.toString()
     }
+
+    private val isURLLenient by lazy {
```

Comment:
I'll leave this up to the lib owners but I think we should remove `isURLLike` or at least have it share the `isSearchTerm` impl because:
- `isURLLike` seems redundant to `isSearchTerm`
- What is `isURLLike` supposed to mean? Without further details, how are consumers supposed to use it confidently? Note that it doesn't look like our code uses it ([query](https://searchfox.org/mozilla-mobile/search?path=&q=isurllike))